### DISCUSSION TITLE
How to add gcc as default LSP for CPP?

### DISCUSSION THREAD

**Discussion by Sarthak1799** [helix-editor/helix#8098] [Q&A]
Currently the default is set to clangd, but clangd does not support bits/stdc++.h and related files. I've always used gcc so would need to set that as default. I see that there is no languages.toml in local AppData (using windows).

Even if I create one, I'm not sure what to put in there as not to bright on the compiler side as well. 
Alternatively, If someone already has gcc configured as default, can they share the data of what to configure in the file?

Any help will be appreciated!

**Comment by pascalkuthe** [comment]
I don't think GCC has a LS. A language server is much more than just something that invokes a compiler. It essentially requires a very specific architecture in the entire compiler frontend (in many programming languages the language server is an entirely independent compiler). AFAIK all c++/c language servers are clang based because other c compilers don't have the necessary architecture.

There might be some kind of comparability mode (potentially just by using the right compliler flafs) but I don't think a GCC based LS exists

**Reply by Sarthak1799** [reply]
I'm not able to use certain headers as clangd probably can't find this (bits/stdc++.h a gcc specific header as it turns out). 
In case anyone in this community who're particularly familiar with cpp might be able to  help out I guess. 
I'm currently looking all over for a solution.

**Comment by Eternal-night-blip** [comment]
I am new in C programming, do you think about using clangd as a language server but using gcc to compile? If you do like this, could do teach me how to do it?

**Comment by TornaxO7** [comment]
There's [`bear`](https://github.com/rizsotto/Bear) which can generate you a `compile_commands.json` which contains all libraries which you include in your program. With this, `clangd` should be able to find `bits/std++.h`.
